catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/MetacardIngestNetworkPlugin.java

```diff
@@ -0,0 +1,174 @@
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PreAuthorizationPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * Conditional attribute adjustments for metacards based on IP, hostname, scheme, and context path.
+ */
+public class MetacardIngestNetworkPlugin implements PreAuthorizationPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardIngestNetworkPlugin.class);
+
+    private static final String CLIENT_INFO_KEY = "client-info";
+
+    private final KeyValueParser keyValueParser;
+
+    private final MetacardServices metacardServices;
+
+    private final AttributeFactory attributeFactory;
+
+    private final MetacardCondition metacardCondition;
+
+    private List<String> newAttributes;
+
+    /**
+     * Default constructor.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            MetacardServices metacardServices) {
+        this(keyValueParser, metacardServices, new AttributeFactory(), new MetacardCondition());
+    }
+
+    /**
+     * Extended constructor with an {@link AttributeFactory} and {@link MetacardCondition} to provide
+     * more fine-grained control.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            MetacardServices metacardServices, AttributeFactory attributeFactory,
+            MetacardCondition metacardCondition) {
+        this.keyValueParser = keyValueParser;
+        this.metacardServices = metacardServices;
+        this.attributeFactory = attributeFactory;
+        this.metacardCondition = metacardCondition;
+    }
+
+    public String getCriteriaKey() {
+        return metacardCondition.getCriteriaKey();
+    }
+
+    public void setCriteriaKey(String criteriaKey) {
+        metacardCondition.setCriteriaKey(criteriaKey);
+    }
+
+    public String getExpectedValue() {
+        return metacardCondition.getExpectedValue();
+    }
+
+    public void setExpectedValue(String expectedValue) {
+        metacardCondition.setExpectedValue(expectedValue);
+    }
+
+    public List<String> getNewAttributes() {
+        return newAttributes;
+    }
+
+    public void setNewAttributes(List<String> newAttributes) {
+        this.newAttributes = newAttributes;
+    }
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        apply(input.getMetacards(), getClientInfoProperties());
+        return input;
+    }
+
+    //region - Passthrough  methods
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
```

Comment:
Default methods can cause issues with OSGi service and backwards compatibility.  If the interface is shared between bundles then both bundles need to be compiled with the same version of the interface.  That means that integrators would need to recompile their bundles any time the default methods are changed.  I would recommend an abstract class instead which would do not cause a backward compatibility issue in OSGi.